python/katana/example_data.py

```diff
@@ -4,66 +4,181 @@
+
+
+    >>> from katana.local import Graph
+    ... graph = Graph(get_rdg_dataset("ldbc_003", as_url=True))
+    """
+    latest_rdg_storage_format_version = get_latest_storage_format_version()
+    return get_rdg_dataset_at_version(rdg_name, latest_rdg_storage_format_version, as_url)
+
+
+def get_csv_dataset(csv_name, as_url=False) -> str:
+    """
+    Get the path to the specified csv dataset in external/test-datasets/csv_datasets
+    """
+
+    rel_path = Path("csv_datasets") / csv_name
+    csv_path = _get_dataset(rel_path)
+    if not csv_path.exists():
+        raise ValueError("csv {} is not available at {}".format(csv_name, csv_path))
+
+    csv_path.resolve()
+    if as_url:
+        return url.Url(csv_path)
 
+    return csv_path
 
-def get_input_as_url(rel_path) -> str:
+
+def get_misc_dataset(misc_rel_path, as_url=False) -> str:
+    """
+    Get the path to the specified misc dataset in external/test-datasets/misc_datasets
+    """
+
+    rel_path = Path("misc_datasets") / misc_rel_path
+    misc_path = _get_dataset(rel_path)
+    if not misc_path.exists():
+        raise ValueError("misc dataset {} is not available at {}".format(misc_rel_path, misc_path))
+
+    misc_path.resolve()
+    if as_url:
+        return url.Url(misc_path)
+
+    return misc_path
+
+
+## functions to access test datasets git repo, not for users of example_data
+def _get_test_datasets_directory(invalidate_cache=False) -> Path:
     """
-    Similar to get_input, but return the graph as file:// URL.
+    *** Don't use this, use get_[rdg,csv,misc]_dataset() ***
+    Locate the test datasets directory
+    if invalidate_cache is passed, and the datasets are not available from the source
+    then the datasets cache will be deleted and re-downloaded
     """
-    path = get_input(rel_path).resolve()
-    return f"file://{path}"
+    test_datasets_path = None
+
+    def validate_path(path: Path):
+        return path.is_dir() and (path / "rdg_datasets").is_dir() and (path / "csv_datasets").is_dir()
+
+    def find_path() -> Path:
+        found_path = None
+        # If SRC_DIR or KATANA_SOURCE_DIR environment is set, just use it
+        if "SRC_DIR" in os.environ:
+            paths_to_check = [Path(os.environ["SRC_DIR"])]
+        elif "KATANA_SOURCE_DIR" in os.environ:
+            paths_to_check = [Path(os.environ["KATANA_SOURCE_DIR"])]
+        else:
+            paths_to_check = list(Path(__file__).parents) + list(Path.cwd().parents)
+            for path in paths_to_check:
+                tmp_path = path / "external" / "test-datasets"
+                if validate_path(tmp_path):
+                    found_path = tmp_path
+
+        return found_path
+
+    def download_datasets(invalidate_cache=False) -> Path:
+        # TODO(emcginnis) it would be really really nice if this got updated automatically
+        # when the submodule ref held by open katana is updated
+        datasets_sha = "14d993df871ddda26964958e271a8bc04d012d40"
```

Comment:
This should be a top level constant, and it should probably check the hash when using the data from the source tree, that would help catch failures to update this, or typos.

We don't yet have a generated Python file to hold build time configuration, but I think we should. This isn't the only place it would be helpful. With such a file, this hardcoded hash would no longer be needed.

Reply:
Good idea, implemented a git sha check on the local test-datasets repo.